frontend/model/contracts/group.js

```diff
@@ -311,8 +312,11 @@ DefineContract({
       // OPTIMIZE: Make this custom validation function
       // reusable accross other future validators
       validate: objectMaybeOf({
+        groupName: x => typeof x === 'string',
+        groupPicture: x => typeof x === 'string',
+        sharedValues: x => typeof x === 'string',
         mincomeAmount: x => typeof x === 'number' && x > 0,
-        mincomeCurrency: x => typeof x === 'string'
+        mincomeCurrency: x => Object.keys(currencies).includes(x)
```

Comment:
This is an `O(n)` operation. You can keep it `O(1)` by doing something along the lines of

```js
x in currencies
```

Reply:
Hum... I don't think that way of handling "versions" is well structure. The way I see it, our code should be retro compatible. So version 1 does:
```
mincomeCurrency: x => !!currencies[x]
```
and version 2 does:

```
mincomeCurrency: x => !!currencies[x] || !!oldCurrencies[x]
```

And so on until we start to deprecate version and "force" users to update their version.

But I'm not an expert on that, so I reverted the change and kept the type of validation.
reverted.